contracts/TownHallZap.sol

```diff
@@ -28,4 +36,61 @@ contract TownHallZap {
             townHall.mint(to);
         }
     }
+
+    // exactInputSingle for ERC20
+    function convertAndMint(address sourceToken, uint256 amountIn, uint24 _fee, address mintTo) external {
+        require(sourceToken != address(0), "sourceToken not address(0)");
+        require(mintTo.code.length == 0, "mintTo is not user");
```

Comment:
This check may not be necessary because it will throw an error in `_mint` function on ERC721 contract:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol#L265

Reply:
fixed